songbird/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/FelixMcFelix/serenity/voice-rework/songbird/songbird.png",
+    html_favicon_url = "https://raw.githubusercontent.com/FelixMcFelix/serenity/voice-rework/songbird/songbird-ico.png"
+)]
+#![deny(missing_docs)]
+//! ![project logo][logo]
+//!
+//! Songbird is an async, cross-library compatible voice system for Discord, written in Rust. The library offers:
+//!  * A (standalone) gateway frontend compatible with [serenity] and [twilight] using the `"gateway"` and `"[serenity/twilight]-[rustls/native]"` features.
+//!  * An optionally standalone driver for voice calls, via the `"driver"` feature.
```

Comment:
The gateway doesn't support any other backend at the moment. It could be made to do so with a `Box<dyn Trait>` in `Shard` etc., but I had to cull the feature creep somewhere.

The driver is entirely standalone. It is possible to run only the driver if you so wish (i.e., you can create a `ConnectionInfo` in your python bot, and somehow pass that into a rust rust program using this crate to roll-your-own-lavalink). I'll explain this further.